src/api/domain/index.ts

```diff
@@ -0,0 +1,201 @@
+import {
+  AccountConnection,
+  EstablishedConnection,
+  WalletConnection,
+} from '../connection/types';
+import { TransactionType, ZkVerifyEvents } from '../../enums';
+import { checkReadOnly, getSelectedAccount } from '../../utils/helpers';
+
+import EventEmitter from 'events';
+import { EventRecord } from '@polkadot/types/interfaces';
+import { RegisterDomainTransactionInfo } from '../../types';
+import { Vec } from '@polkadot/types';
+import { VerifyOptions } from '../../session/types';
+import { handleTransaction } from '../../utils/transactions';
+
+export const registerDomain = async (
+  connection: AccountConnection | WalletConnection | EstablishedConnection,
+  aggregationSize: number,
+  queueSize: number = 16,
+  emitter: EventEmitter,
+): Promise<number> => {
+  if (aggregationSize > 128) throw new Error('aggregationSize must be <= 128');
+  if (queueSize > 16) throw new Error('queueSize must be <= 16');
+
+  checkReadOnly(connection);
+
+  const { api } = connection;
+  let selectedAccount;
+
+  if ('accounts' in connection) {
+    selectedAccount = getSelectedAccount(connection);
+  }
+
+  const registerExtrinsic = api.tx.aggregate.registerDomain(
+    aggregationSize,
+    queueSize,
+  );
+
+  try {
+    const transactionResult = await (async () => {
+      if (selectedAccount) {
+        return await handleTransaction(
+          api,
+          registerExtrinsic,
+          selectedAccount,
+          undefined, // Not sure about this, should we be using a signer here?
+          emitter,
+          {} as VerifyOptions, // Do we need verify options for domain registration?
```

Comment:
It looks like `handleTransaction` was written for proof-related transactions. The function might need a slight refactor to handle the empty object